packages/button/components/ButtonBase.tsx

```diff
@@ -91,115 +91,102 @@ export interface ButtonBaseProps extends ButtonProps {
-    url,
-    openInNewTab,
-    ...other
-  } = props;
-
-  const buttonClassName = cx(
-    buttonReset,
-    button(appearance),
-    buttonBase,
-    textWeight("medium"),
-    className,
-    {
-      [fullWidthButton]: isFullWidth,
-      [buttonInverse(appearance)]: isInverse,
-      [getMutedButtonStyles(appearance)]: disabled || isProcessing,
-      [getInverseMutedButtonStyles(appearance)]:
-        (disabled || isProcessing) && isInverse
-    }
-  );
-
-  const handleClick = (e: React.SyntheticEvent<HTMLElement>) => {
-    if (!disabled && onClick) {
-      onClick(e);
-    }
-  };
-
-  const getIconStart = icon => {
-    return (
-      <span className={cx(flexItem("shrink"), display("inherit"))}>
-        <IconPropAdapter icon={icon} size="xs" color="inherit" />
-      </span>
-    );
-  };
-
-  const getIconEnd = icon => {
-    return (
-      Boolean(icon) && (
+const ButtonContent = ({ iconStart, iconEnd, isProcessing, children }) => {
+  return iconStart || iconEnd ? (
+    <span className={flex({ align: "center", justify: "center" })}>
+      {iconStart && (
+        <span className={cx(flexItem("shrink"), display("inherit"))}>
+          <IconPropAdapter icon={iconStart} size="xs" color="inherit" />
+        </span>
+      )}
+      {children && (
+        <span
+          className={cx(flexItem("shrink"), padding("left", "xs"), {
+            [processingTextStyle]: isProcessing
+          })}
+        >
+          {children}
+        </span>
+      )}
+      {iconEnd && (
         <span
           className={cx(
             flexItem("shrink"),
             display("inherit"),
             padding("left", "xxs")
           )}
         >
-          <IconPropAdapter icon={icon} size="xs" color="inherit" />
+          <IconPropAdapter icon={iconEnd} size="xs" color="inherit" />
         </span>
-      )
-    );
-  };
-
-  const getButtonContent = () => {
-    const { iconStart, iconEnd, isProcessing, children } = props;
+      )}
+    </span>
+  ) : (
+    <span className={isProcessing ? processingTextStyle : ""}>{children}</span>
+  );
+};
 
-    return iconStart || iconEnd ? (
-      <span className={flex({ align: "center", justify: "center" })}>
-        {iconStart && getIconStart(iconStart)}
-        {children && (
-          <span
-            className={cx(flexItem("shrink"), padding("left", "xs"), {
-              [processingTextStyle]: isProcessing
-            })}
-          >
-            {children}
-          </span>
-        )}
-        {iconEnd && getIconEnd(iconEnd)}
-      </span>
-    ) : (
-      <span className={isProcessing ? processingTextStyle : ""}>
-        {children}
-      </span>
+const ButtonNode = React.forwardRef(
+  (
+    {
+      appearance,
+      children,
+      className,
+      disabled,
+      iconStart,
+      iconEnd,
+      isInverse,
+      isProcessing,
+      isFullWidth,
+      onClick,
+      type = "button",
+      url,
+      openInNewTab,
+      ariaHaspopup,
+      ...other
```

Comment:
I found another console warning while testing this out:

> Warning: Invalid ARIA attribute `ariaLabel`. Did you mean `aria-label`?

It's a minor fix if you were happy to add it to this PR. You would need to destructure `ariaLabel` here, and then add a `aria-label` prop in the button element on line 195 and set `ariaLabel` as the value.